mgmtsystem/azure.py

```diff
@@ -96,21 +98,42 @@ def stop_vm(self, vm_name, resource_group=None, shutdown=False):
             self.run_script(
                 "Stop-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\" -Force"
                 .format(resource_group or self.resource_group, vm_name).strip())
+            self.wait_vm_stopped(vm_name, resource_group)
         else:
             self.logger.info("Azure VM {} is already stopped".format(vm_name))
 
+    def suspend_vm(self, vm_name, resource_group=None):
+        if self.is_vm_running(vm_name, resource_group or self.resource_group):
+            self.logger.info("Attempting to Suspend Azure VM {}".format(vm_name))
+            self.run_script(
+                "Stop-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\" -Force -StayProvisioned"
+                .format(resource_group or self.resource_group, vm_name).strip())
+            self.wait_vm_suspended(vm_name, resource_group)
+        else:
+            self.logger.info("Azure VM {} is already suspended or stopped".format(vm_name))
+
     def wait_vm_running(self, vm_name, resource_group=None, num_sec=300):
         wait_for(
             lambda: self.is_vm_running(vm_name, resource_group or self.resource_group),
             message="Waiting for Azure VM {} to be running.".format(vm_name),
             num_sec=num_sec)
 
+    def wait_vm_steady(self, vm_name, resource_group=None, num_sec=300):
+        self.logger.info("All states are steady in Azure. {}".format(vm_name))
```

Comment:
If you look at the above snippet, there is almost 15 minutes of power toggling go on which isn't necessary for Azure.  When I first ran test_suspend, it through an exception on wait_vm_steady as it was not implemented.  That was the error I was trying to correct.  I first corrected the mgmt system, and later just updated vm.py so that the logic is skipped all together.  So as I said above, I think this additional override will prevent future errors, but for now, I'm happy to delete it and make sure we never call it.  A quick file search shows it's not really used all that much in the code base.